cli/exchange/nmp.go

```diff
@@ -155,3 +170,87 @@ func NMPRemove(org, credToUse, nmpName string, force bool) {
 		msgPrinter.Println()
 	}
 }
+
+func NMPListNodes(org, credToUse, nmpName string) {
+	cliutils.SetWhetherUsingApiKey(credToUse)
+
+	var nmpOrg string
+	nmpOrg, nmpName = cliutils.TrimOrg(org, nmpName)
+
+	if nmpName == "*" {
+		nmpName = ""
+	}
+
+	// get message printer
+	msgPrinter := i18n.GetMessagePrinter()
+
+	// store list of compatible nodes in map indexed by NMP's in the exchange
+	compatibleNodeMap := make(map[string][]string)
+
+	var nmpList exchange.ExchangeNodeManagementPolicyResponse
+	var output string
+	httpCode := cliutils.ExchangeGet("Exchange", cliutils.GetExchangeUrl(), "orgs/"+nmpOrg+"/managementpolicies"+cliutils.AddSlash(nmpName), cliutils.OrgAndCreds(org, credToUse), []int{200, 404}, &nmpList)
+	if httpCode == 404 && nmpName != "" {
+		cliutils.Fatal(cliutils.NOT_FOUND, msgPrinter.Sprintf("NMP %s not found in org %s", nmpName, nmpOrg))
+	} else if httpCode == 404 {
+		output = "{}"
+	} else {
+		for nmp, nmpPolicy := range nmpList.Policies {
+			nodes := determineCompatibleNodes(org, credToUse, nmpName, nmpPolicy)
+			compatibleNodeMap[nmp] = nodes
+		}
+		output = cliutils.MarshalIndent(compatibleNodeMap, "management nmp list --nodes")
+	}
+	fmt.Printf(output)
+	msgPrinter.Println()
+}
+
+func NMPListStatus(org, credToUse, nmpName string) {
+	// cliutils.SetWhetherUsingApiKey(credToUse)
+
+	// var nmpOrg string
+	// nmpOrg, nmpName = cliutils.TrimOrg(org, nmpName)
+
+	// if nmpName == "*" {
+	// 	nmpName = ""
+	// }
+
+	// get message printer
+	// msgPrinter := i18n.GetMessagePrinter()
+
+	// var nmpList exchange.ExchangeNodeManagementPolicyResponse
+	// httpCode := cliutils.ExchangeGet("Exchange", cliutils.GetExchangeUrl(), "orgs/"+nmpOrg+"/managementpolicies"+cliutils.AddSlash(nmpName), cliutils.OrgAndCreds(org, credToUse), []int{200, 404}, &nmpList)
+}
+
+func determineCompatibleNodes(org, credToUse, nmpName string, nmpPolicy exchangecommon.ExchangeNodeManagementPolicy) []string {
+	var nmpOrg string
+	nmpOrg, nmpName = cliutils.TrimOrg(org, nmpName)
+	
+	// get node(s) name(s) from the Exchange
+	var resp ExchangeNodes
+	cliutils.ExchangeGet("Exchange", cliutils.GetExchangeUrl(), "orgs/"+nmpOrg+"/nodes", cliutils.OrgAndCreds(org, credToUse), []int{200, 404}, &resp)
+
+	compatibleNodes := []string{}
+	for nodeNameEx, node := range resp.Nodes {
```

Comment:
Add a check to make sure that the node is registered.  If `node.PublickKey != ""`, then it is registered.